### ISSUE TITLE
Support for `ScaleInvariant` Loss

### ISSUE THREAD

**Issue by awsaf49** [keras-team/keras#17719]
**System information**.

TensorFlow version (you are using): 2.11.0
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.
The Scale-Invariant (SI) loss is very popular in the field of Depth Estimation introduced in [this paper](https://proceedings.neurips.cc/paper/2014/file/7bccfde7714a1ebadf06c5f4cea752c1-Paper.pdf). Almost all papers in Depth Estimation use this loss for the training models. The mathematical equation for this loss is:

$$
\mathcal{L}_{\text {pixel }}=\alpha \sqrt{\frac{1}{T} \sum_{i} g_i^2-\frac{\lambda}{T^2}\left(\sum_i g_i\right)^2}
$$

Where $d_i$ is the ground truth depth, $\hat{d}_i$  is the estimated depth, $T$ represents the number of pixels with valid ground truth values and $g_i$ is computed as $g_i=\log_e(\hat{d}_i)-\log_e(d_i)$. The parameters $\lambda$ and $\alpha$ are hyperparameter.

**Will this change the current api? How?** N/A

**Who will benefit from this feature?**
Scale Invariant loss is a regression loss that can be applied to any pixel-wise regression task. Specifically, this loss is widely used in Depth Estimation thus anyone who is working in this field will be benefited from it.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): Yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

Here is a candidate solution

```py
class ScaleInvariant(tf.keras.losses.Loss):
    def __init__(self,
                 alpha=10.0,
                 lamda=0.85,
                ):
        super().__init__()
        self.alpha = alpha
        self.lamda = lamda

    def call(self, y_true, y_pred):
        # create vaid mask
        mask = self.get_mask(y_true)
        # compute loss
        loss = self.compute_loss(y_true, y_pred, mask)
        return loss
```

**Comment by awsaf49** [comment]
@sushreebarsa any update on this?

**Comment by rchao** [comment]
Thanks for the request - with the citation I think we can move on with an PR for us to review. 

cc'ing @fchollet in case there's opinion.

**Comment by rchao** [comment]
cc'ing @rchao

**Comment by SuryanarayanaY** [comment]
Hello, Thank you for reporting an issue.

We're currently in the process of migrating the new Keras 3 code base from keras-team/keras-core to keras-team/keras.
Consequently, This issue may not be relevant to the Keras 3 code base . After the migration is successfully completed, feel free to reopen this Issue at keras-team/keras if you believe it remains relevant to the Keras 3 code base. If instead this Issue is a bug or security issue in legacy tf.keras, you can instead report a new issue at [keras-team/tf-keras](https://github.com/keras-team/tf-keras/issues), which hosts the TensorFlow-only, legacy version of Keras.

To know more about Keras 3, please read https://keras.io/keras_core/announcement/
